pkg/server/slicerouter_config.go

```diff
@@ -112,21 +112,19 @@ func vl3DeleteRouteInVpp(dstIP string, nextHopIP string) error {
 	return sendConfigToVppAgent(vppconfig, true)
 }
 
-func vl3InjectRouteInKernel(dstIP string, nextHopIP string) error {
+func vl3InjectRouteInKernel(dstIP string, nextHopIPSlice []*netlink.NexthopInfo) error {
 	_, dstIPNet, err := net.ParseCIDR(dstIP)
 	if err != nil {
 		return err
 	}
-	gwIP := net.ParseIP(nextHopIP)
 
-	route := netlink.Route{Dst: dstIPNet, Gw: gwIP}
+	route := netlink.Route{Dst: dstIPNet, MultiPath: nextHopIPSlice}
 
-	if err := netlink.RouteReplace(&route); err != nil {
-		logger.GlobalLogger.Errorf("Route add failed in kernel. Dst: %v, NextHop: %v, Err: %v", dstIPNet, gwIP, err)
+	if err := netlink.RouteAddEcmp(&route); err != nil {
```

Comment:
I think you will have to use route replace here instead of add. If one of the nexthops of the ecmp needs to be removed, we would have to do a route replace and retain only the nexthops that are needed. Please check with @kon3m as he is working on deleting one of the nexthops. You can leave this as is and have @kon3m do it as part of his story or you can do it as part of this PR itself. Please coordinate with him.